### ISSUE TITLE
Wrong 'self' in the onmouseenter/onmouseleave.

### ISSUE THREAD

**Issue by actboy168** [yue/yue#16]
os:win10

![1](https://user-images.githubusercontent.com/1836844/36825275-73bdff36-1d41-11e8-97a8-d2a45e7f48b7.gif)

Error message:
```
[0301/110440.107:ERROR:callback_internal.h(219)] Error when calling lua function: error converting arg at index 2 from nil to yue.Color
```

```lua
local gui = require 'yue.gui'

local window = {}

local win = gui.Window.create { }
function win.onclose()
    gui.MessageLoop.quit()
end

local view = gui.Container.create()

local colors = { '#00ADD9', '#00AD3C', '#D9A33C' }
for i = 1, 3 do
    local color1 = colors[i]
    local color2 = ('#%06X'):format(tonumber('0x' .. color1:sub(2)) + 0x101010)
    local btn = gui.Button.create('')
    btn:setbackgroundcolor(color1)
    btn._backgroundcolor1 = color1
    btn._backgroundcolor2 = color2
    function btn:onmouseleave()
        print('onmouseleave', self, self._backgroundcolor1)
        self:setbackgroundcolor(self._backgroundcolor1)
    end
    function btn:onmouseenter()
        print('onmouseenter', self, self._backgroundcolor2)
        self:setbackgroundcolor(self._backgroundcolor2)
    end

    btn:setstyle { Margin = 10, FlexGrow = 1 }
    view:addchildview(btn)
end

win:setcontentview(view)
win:setcontentsize { width = 400, height = 600 }
win:center()
win:activate()

gui.MessageLoop.run()

```

**Comment by zcbenz** [comment]
The `view:addchildview` method does not keep a reference to the passed lua variable, so the lua variable`btn` gets garbage collected after some time and gets recreated when you access it again.

Currently you can work around this bug by storing `btn` in a global table to prevent it from being garbage collected.

**Comment by zcbenz** [comment]
Fixed with 58c90d84fe6abd6964ecdb0c6b53a143ba8aeabd.
